ui/raidboss/emulator/EmulatorCommon.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+class EmulatorCommon {
+  static getTimestampFromLogLine(day, line) {
+    let result = EmulatorCommon.logLineRegex.exec(line);
+    return +new Date(day + ' ' + result.groups.lineTimestamp);
+  }
+}
+
+EmulatorCommon.logLineRegex = /^\[(?<lineTimestamp>[^\]]+)\] ([0-9A-Z]+):(.*)$/i;
+
+EmulatorCommon.SealRegexes = {
```

Comment:
Same problem as the `EventDetailsRegex`, the existing regexes don't capture all the fields. In this case it's the timestamp. Since the emulator needs to chunk through thousands of lines all at once, every additional regex we need to run (to e.g. pull the timestamp from the line) means a significant increase in processing time.